### DISCUSSION TITLE
docker image with custom setup

### DISCUSSION THREAD

**Discussion by r0ckyjoe** [exadel-inc/CompreFace#1027] [Q&A] [answered]
Hello team,
I am wondering if it is possible to create a customized docker image of the compreface stack with a recognition app and a user already created and thus the api-key, so that once the docker-compose stack is up the service is ready to use without the need to configure it.
I was thinking to start a blank compose-stack, create the user and the app, then take the stack down and create a docker image with what I have.

Is there a better way to do so? 

Thanks,
~RJ

**Comment by jvrobert** [comment, marked as answer]
One way might be to use a script in a command: section to create them using the API, that would work on any image you start up.

**Reply by r0ckyjoe** [reply]
Thanks! When you say via API in a script, do you mean with a SDK?
Are you aware of any examples of such a way to configure compreface I can have a look at?

~RJ

**Reply by jvrobert** [reply]
I took a quick glance at the API docs and don't see a way to add an app/service but maybe I just missed it. But looking at the REST calls the web client makes, it looks trivial so you could hack the API to make those calls pretty easily, or use CURL or something (would have to authenticate).
I just did a quick test with curl --cookie "CFSESSION=MYSESSIONID" -L http://mycompreface.server.com/admin/user/me and it worked, the session ID is the value I pulled from the web browser (for real script you'd login and get one). So you could do that with an appropriate POST, or maybe one of the devs has an easier idea.

**Reply by pospielov** [reply]
CompreFace has predefined demo services:
Face recognition service: 00000000-0000-0000-0000-000000000002
Face detection service: 00000000-0000-0000-0000-000000000003
Face verification service: 00000000-0000-0000-0000-000000000004
So if you don't need several Facial recognition services, you can just use those ones.
Also, the Face recognition service comes with predefined face examples, so you need to delete them first with REST API.

**Reply by r0ckyjoe** [reply]
Thank you both!
